Fix and update shikiji core (#10063)
withastro/astro: .changeset/poor-bees-juggle.md

---
"astro": patch
---

> Moves `shikiji-core` as a dependency to prevent type errors

Reviewer: ```suggestion
Moves `shikiji-core` from `devDependencies` to `dependencies` to prevent type errors
```
